gpcontrib/arenadata_toolkit/src/arenadata_toolkit_guc.c

```diff
@@ -0,0 +1,295 @@
+#include "arenadata_toolkit_guc.h"
+
+
+#include "cdb/cdbvars.h"
+#include "catalog/objectaccess.h"
+#include "catalog/pg_db_role_setting.h"
+#include <limits.h>
+#include "utils/guc.h"
+#include "tf_shmem.h"
+
+#define DEFAULT_BLOOM_SIZE 1000000
+#define DEFAULT_DB_TRACK_COUNT 5
+#define DEFAULT_IS_TRACKED false
+#define DEFAULT_DROPS_COUNT 100000
+#define DEFAULT_TRACKED_SCHEMAS "public,arenadata_toolkit,pg_catalog,pg_toast,pg_aoseg,information_schema"
+#define DEFAULT_GET_FULL_SNAPSHOT_ON_RECOVERY true
+#define DEFAULT_TRACKED_REL_STORAGES "h,a,c"
+#define DEFAULT_TRACKED_REL_KINDS "r,i,t,m,o,b,M"
+#define DEFAULT_NAPTIME 60
+
+#define MIN_BLOOM_SIZE 1
+#define MIN_DB_TRACK_COUNT 1
+#define MIN_DROPS_COUNT 1
+#define MIN_NAPTIME 1
+
+#define MAX_BLOOM_SIZE 128000000
+#define MAX_DB_TRACK_COUNT 1000
+#define MAX_DROPS_COUNT 1000000
+#define MAX_NAPTIME OID_MAX & 0x7FFFFFFF
+
+int			bloom_size = DEFAULT_BLOOM_SIZE;
+int			db_track_count = DEFAULT_DB_TRACK_COUNT;
+bool		is_tracked = DEFAULT_IS_TRACKED;
+bool		get_full_snapshot_on_recovery = DEFAULT_GET_FULL_SNAPSHOT_ON_RECOVERY;
+int			drops_count = DEFAULT_DROPS_COUNT;
+char	   *tracked_schemas = DEFAULT_TRACKED_SCHEMAS;
+char	   *tracked_rel_storages = DEFAULT_TRACKED_REL_STORAGES;
+char	   *tracked_rel_kinds = DEFAULT_TRACKED_REL_KINDS;
+int			tracking_worker_naptime_sec = DEFAULT_NAPTIME;
+
+static bool is_tracked_unlocked = false;
+static bool is_get_full_snapshot_on_recovery_unlocked = false;
+static bool is_schemas_unlocked = false;
+static bool is_relkinds_unlocked = false;
+static bool is_relstorages_unlocked = false;
+
+void
+tf_guc_unlock_tracked_once(void)
+{
+	if (!is_tracked_unlocked)
+		is_tracked_unlocked = true;
```

Comment:
why not just:
```suggestion
	is_tracked_unlocked = true;
```
The same below.